WalletWasabi/Crypto/ZeroKnowledge/Credential.cs

```diff
@@ -0,0 +1,22 @@
+﻿using NBitcoin.Secp256k1;
+using WalletWasabi.Crypto.Groups;
+
+namespace WalletWasabi.Crypto.ZeroKnowledge
+{
+	public class Credential
+	{
+		public Credential(Scalar amount, Scalar randomness, MAC mac)
+		{
+			Amount = amount;
+			Randomness = randomness;
+			Mac = mac;
+		}
+
+		public Scalar Amount { get; } // could consider uint, but only really makes sense for credential requests because of range proofs
```

Comment:
the comment was for you not for the code ;-)

```suggestion
		public Scalar Amount { get; }
```